### PR TITLE
Partition some pools by core

### PR DESCRIPTION
**Pull request by sebastienros**
Benchmarks to follow

### CODE REVIEW

**Comment by BrennanConroy** [comment]
Should probably measure the two different pool changes independently and then together.

**Comment by sebastienros** [comment]
@BrennanConroy The Sockets change is only measurable when the memory block one is fixed, otherwise it's hidden. I can run one benchmark with memory / memory + socket at least to show it matters, but will do add the extended benchmarks only using the two changes, otherwise it's too many combinations.

**Comment by BrennanConroy** [comment]
> The Sockets change is only measurable when the memory block one is fixed, otherwise it's hidden.

I see, makes sense 👍

**Comment by kunalspathak** [comment]
How does the benchmark results look like?

**Comment by samsp-msft** [comment]
@sebastienros - should we have the YARP benchmarks running on this hardware?

**Review by adamsitnik — COMMENTED** [review]
I would love the see the JSON Platform results using these changes (it's the best benchmark for testing max throughput)

File: src/Shared/Buffers.MemoryPool/PinnedBlockMemoryPool.cs [hunk 1]
```diff
@@ -58,7 +67,10 @@ public override IMemoryOwner<byte> Rent(int size = AnySize)
             MemoryPoolThrowHelper.ThrowObjectDisposedException(MemoryPoolThrowHelper.ExceptionArgument.MemoryPool);
         }
 
-        if (_blocks.TryDequeue(out var block))
+        
+        var partition = Thread.GetCurrentProcessorId() % _queues.Length;
+        
+        if (_queues[partition].TryDequeue(out var block))
```
[the remark this answers is not in the document]
Comment (kouvel) [reply src/Shared/Buffers.MemoryPool/PinnedBlockMemoryPool.cs:73]: It likely would be better to assign a pooled item with a partition and return it to the same partition it was taken from. I had seen this issue before and recently again in testing with thread pool changes along with this change, where there is still a fair bit of allocation. The issue is that currently, some threads rent buffers and likely other threads return them, so the threads renting buffers will eventually run out and will have to allocate. Returning the buffers (or pooled items similarly for `SocketSenderPool` to the same queue) would maintain some balance between the caches.

**Comment by kouvel** [comment]
@sebastienros, any update on this PR? Would like to have this for .NET 7 for further experiments.

**Comment by sebastienros** [comment]
Will come back to it very soon, as soon as I am done with output caching (#41037)

**Comment by japui** [comment]
Particionar alguns pools por núcleo

**Comment by sebastienros** [comment]
Replaced by https://github.com/dotnet/aspnetcore/pull/42237

**Pull request closed without merging**